server_test.go

```diff
@@ -16,69 +16,57 @@ import (
 )
 
 func TestStubServer(t *testing.T) {
-	server := getStubServer(t)
-
-	req := httptest.NewRequest("POST", "https://stripe.com/v1/charges",
-		bytes.NewBufferString("amount=123&currency=usd"))
-	req.Header.Set("Authorization", "Bearer sk_test_123")
-	w := httptest.NewRecorder()
-	server.HandleRequest(w, req)
-
-	resp := w.Result()
-	body, err := ioutil.ReadAll(resp.Body)
-	assert.NoError(t, err)
-
+	resp, body := sendRequest(t, "POST", "/v1/charges",
+		"amount=123&currency=usd", getDefaultHeaders())
 	assert.Equal(t, http.StatusOK, resp.StatusCode)
 
 	var data map[string]interface{}
-	err = json.Unmarshal(body, &data)
+	err := json.Unmarshal(body, &data)
 	assert.NoError(t, err)
 	_, ok := data["id"]
 	assert.True(t, ok)
 }
 
-func TestStubServer_SetsSpecialHeaders(t *testing.T) {
-	server := getStubServer(t)
+func TestStubServerError(t *testing.T) {
+	resp, body := sendRequest(t, "GET", "/a", "", nil)
+	assert.Equal(t, http.StatusUnauthorized, resp.StatusCode)
 
-	// Does this regardless of endpoint
-	req := httptest.NewRequest("GET", "https://stripe.com/", nil)
-	w := httptest.NewRecorder()
-	server.HandleRequest(w, req)
+	var data map[string]interface{}
+	err := json.Unmarshal(body, &data)
+	assert.NoError(t, err)
+	errorInfo, present := data["error"].(map[string]interface{})
```

Comment:
Mind renaming `present` to `ok` here? It's a pretty common convention in Go.

Reply:
fixed